### ISSUE TITLE
catboost inverts the link for Logloss but not for Poisson or Tweedie

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/tidypredict#356]
Found by the ignored-fit-options battery.

## What happens

`R/model-catboost.R:3-14` puts `Poisson` and `Tweedie` in `catboost_identity_objectives`, so `tidypredict_fit()` returns the raw formula value on the log scale. `Logloss` and `CrossEntropy` are in `catboost_sigmoid_objectives` and get a logistic applied.

Those two choices follow different conventions, and only one can match a given `catboost.predict()` call:

| Objective | tidypredict | matches `RawFormulaVal` (catboost's **default**) | matches `Probability` / `Exponent` |
| --- | --- | --- | --- |
| `Poisson` | identity | **yes, 0 difference** | no, 1.457 off `Exponent` |
| `Logloss` | sigmoid | no | **yes, 0 difference** |

So a user who fits `Logloss` gets probabilities, and a user who fits `Poisson` gets log-counts. Neither is wrong on its own terms; together they are inconsistent, and nothing documents which convention applies to which objective.

`xgboost` takes the other convention throughout: `apply_xgb_objective()` (`R/model-xgboost.R:327`) applies `exp()` for `count:poisson`, `reg:tweedie` and `reg:gamma`, matching what `predict()` returns for those objectives. So the same fitted model shape gets a link inverted through the xgboost path and not through the catboost path.

## Measured

20-iteration depth-4 model, Poisson objective, 200 rows:

```
RawFormulaVal  max abs diff vs tidypredict 0
Exponent       max abs diff vs tidypredict 1.457
Probability    max abs diff vs tidypredict 0.652
```

Also swept and clean: `Logloss` (0), `MAE` (0), `boosting_type = "Ordered"` (0).

## The decision

Which convention should catboost follow? Either is arguable:

- **Invert the link for `Poisson` and `Tweedie` too**, matching xgboost and giving predictions on the response scale. Diverges from `catboost.predict()`'s default, as `Logloss` already does.
- **Leave the raw value and drop the sigmoid from `Logloss`**, matching `catboost.predict()`'s default everywhere. Consistent, but changes existing behaviour for the far more common objective and would surprise anyone using it for classification.

The first looks right to me, since the sigmoid for `Logloss` has presumably been relied on, but it is a behaviour change either way and worth deciding deliberately. Whichever is chosen, the objective-to-link mapping should be documented, since it cannot be inferred from `catboost.predict()`'s own default.

**Comment by EmilHvitfeldt** [comment]
A third option, which I think is better than either of the two above.

The framing of the issue was wrong in one respect: it treated this as tidypredict following catboost's convention versus tidypredict following xgboost's. But `R/model-catboost.R` already inverts the link for four of its six supported objective families, not just for `Logloss`:

- `Logloss`, `CrossEntropy` -> logistic (`R/model-catboost.R:741`)
- `MultiClass` -> softmax (`:609`)
- `MultiClassOneVsAll` -> per-class logistic (`:612`)
- `Poisson`, `Tweedie` -> identity

None of the first four match `catboost.predict()`'s `RawFormulaVal` default. So "match `catboost.predict()`'s default everywhere" is not really available: taken seriously it means stripping the softmax off `MultiClass` and returning raw per-class scores, which nobody wants and which no other multiclass model in the package does.

That leaves `Poisson` and `Tweedie` as the two objectives out of step, both with the rest of the catboost parser and with the package-wide convention of returning predictions on the response scale. So it is not a coin flip between two defensible conventions: option 1 is the only one consistent with the code as it stands, and the change is narrow, `exp()` for those two objectives.

Still a behaviour change, so still worth a deliberate decision, but the decision is smaller than the issue made it look.
